ml/src/test/scala/ai/h2o/sparkling/ml/models/H2OMOJOPipelineModelTestSuite.scala

```diff
@@ -107,88 +89,88 @@ class H2OMOJOPipelineModelTestSuite extends FunSuite with SparkTestContext with
     }
   }
 
-  test("Testing dataset is missing one of feature columns") {
-    val schema = spark.read
-      .option("header", "true")
-      .csv("examples/smalldata/prostate/prostate.csv")
-      .drop("AGE")
-      .schema
-    val mojo = H2OMOJOPipelineModel.createFromMojo(
-      this.getClass.getClassLoader.getResourceAsStream("mojo2data/pipeline.mojo"),
-      "prostate_pipeline.mojo")
+  test("Testing dataset is missing one of the feature columns") {
+    val schema = prostateTestData.drop("AGE").schema
     val rdd = sc.parallelize(Seq(Row("1", "0", "1", "2", "1", "1.4", "0", "6")))
     val testingDF = spark.createDataFrame(rdd, schema)
 
-    val predictionsDF = mojo.transform(testingDF)
+    val predictionsDF = prostateMojoPipeline.transform(testingDF)
 
-    // materialize the frame to see that it is passing
-    predictionsDF.collect()
+    noException should be thrownBy predictionsDF.collect()
   }
 
-  test("Testing dataset has an extra feature column") {
-    val schema = spark.read
-      .option("header", "true")
-      .csv("examples/smalldata/prostate/prostate.csv")
-      .withColumn("EXTRA", lit("extra"))
-      .schema
-    val mojo = H2OMOJOPipelineModel.createFromMojo(
-      this.getClass.getClassLoader.getResourceAsStream("mojo2data/pipeline.mojo"),
-      "prostate_pipeline.mojo")
-    val rdd = sc.parallelize(Seq(Row("1", "0", "65", "1", "2", "1", "1.4", "0", "6", "8")))
-    val testingDF = spark.createDataFrame(rdd, schema)
+  test("Testing dataset having an extra column should give the same prediction") {
+    val rdd = sc.parallelize(Seq(Row("1", "0", "65", "1", "2", "1", "1.4", "0", "6")))
+    val recordWithoutAdditionalColumn = spark.createDataFrame(rdd, prostateTestData.schema)
+    val recordWithAdditionalColumn = recordWithoutAdditionalColumn.withColumn("extraCol", lit("8"))
 
-    val predictionsDF = mojo.transform(testingDF)
+    val prediction = prostateMojoPipeline.transform(recordWithoutAdditionalColumn).select(PredictionCol).collect().head
+    val predictionWithAdditionalColumn =
+      prostateMojoPipeline.transform(recordWithAdditionalColumn).select(PredictionCol).collect().head
 
-    // materialize the frame to see that it is passing
-    predictionsDF.collect()
+    prediction shouldEqual predictionWithAdditionalColumn
+  }
+
+  test("Testing dataset with column order mixed should give the same prediction") {
+    val rdd = sc.parallelize(Seq(Row("1", "0", "65", "1", "2", "1", "1.4", "0", "6")))
+    val record = spark.createDataFrame(rdd, prostateTestData.schema)
+    val recordWithDifferentColumnOrder = record.selectExpr(record.columns.reverse: _*).toDF()
+    record.columns shouldNot equal(recordWithDifferentColumnOrder.columns)
+    record.columns.sorted shouldEqual recordWithDifferentColumnOrder.columns.sorted
+
+    val prediction = prostateMojoPipeline.transform(record).select(PredictionCol).collect().head
+    val predictionWithDifferentColumnOrder =
```

Comment:
In the test like this, we need to start doing assertions on multiple rows and make sure the predicted values are not the same.